posts/2023-XX-YY-boa-release-17.md

```diff
@@ -0,0 +1,237 @@
+
+With constant folding expression optimization, the compiler analyzes expressions involving constants and replaces them with their computed results. This process allows the compiler to transform arithmetic operations, comparisons, and logical expressions into simplified forms, removing unnecessary runtime computations. By eliminating these computations, the optimized program benefits from reduced execution time and improved overall performance.
+
+#### Object Shapes (Hidden classes)
+
+Hidden classes (in boa called "shapes" to avoid confusion with JavaScript classes) is an optimization technique that aims to improve the performance of dynamically-typed programming languages. This optimization works by analyzing the patterns of object creation and dynamically generating hidden classes to represent similar object structures.
+
+In dynamically-typed languages, objects are usually a bag of properties (key and value pairs) that are modified at runtime, which can lead to inefficiencies in terms of memory usage. Hidden classes optimization addresses this by organizing object structures more efficiently, reducing memory overhead, it also unlocks the posability for other optimizations, such as inline caching, which optimizes property access.
+
+The shapes create a trasition tree, where the trasitions are property and prototype changes starting from the root shape (no properties, no prototype).
+
+```JavaScript
+let o = {} // Shape 1: prototype `Object.prototype`
+o.a = 10   // Shape 2: prototype `Object.prototype` and propeties: 'a'
+o.b = 20   // Shape 3: prototype `Object.prototype` and propeties: 'a', 'b'
+
+let o2 = { a: 30; } // Shape 2: prototype `Object.prototype` and propeties: 'a'
+o2.d = 50           // Shape 4: prototype `Object.prototype` and propeties: 'a', 'd' -- fork from shape 2
+```
+
+When creating objects with the same property names, it's best to create them in the same order, this ensures that the objects share the same shape.
+
+For a more in depth explanation of how shared shapes work in boa see `shapes.md` [here](https://github.com/boa-dev/boa/blob/main/docs/shapes.md).
+
+TO-DO: explain the enhancements of new optimizations (constant folding, inline catching, hidden classes...)
+
+### Debug object
+
+If you're debugging your JavaScript code with the Boa CLI interface, you can use the new `$boa` object. If you want to use it,
+you will need to run the Boa CLI / REPL with the `--debug-object` command line flag.
+
+The `$boa` debug object is divided into modules, so that you can trigger the garbage collection with `$boa.gc.collect()`, or
+get the bytecode of a function by running `$boa.function.bytecode(fn_name)`. You can also trace function invocations, handle
+compiler optimizations, limits and object shapes.
+
+You can find all the documentation [here][debug_object].
+
+[debug_object]: https://github.com/boa-dev/boa/blob/main/docs/boa_object.md
+
+### Modules
```

Comment:
Should the `Modules` section be moved farther up in the post? It's a pretty big update.